File: sdk/dotnet/src/xjustiz.core-dotnet/Models/Codes/Telekommunikationsart/TelekommunikationsartCodeLists.cs
namespace xjustiz.core_dotnet.Models.Codes.Telekommunikationsart;

/// <summary>
/// Versionslisten für <see cref="TelekommunikationsartCode"/> aus <a href="https://www.xrepository.de/details/urn:xoev-de:xjustiz:codeliste:gds.telekommunikationsart">XRepository: Codelisten GDS.Telekommunikationsart</a>.<br/>
/// <u><b>Telecommunication type code lists:</b></u> for <see cref="TelekommunikationsartCode"/> from <a href="https://www.xrepository.de/details/urn:xoev-de:xjustiz:codeliste:gds.telekommunikationsart">XRepository: Code lists GDS.TelecommunicationType</a>.
/// </summary>
public sealed class TelekommunikationsartCodeLists : ICodeListCollection<TelekommunikationsartCode>
{
    /// <summary>
    /// Telekommunikationsart Codes Version <b>2.2</b> aus <a href="https://www.xrepository.de/api/xrepository/urn:xoev-de:xjustiz:codeliste:gds.telekommunikationsart_2.2:technischerBestandteilGenericode">XRepository: GDS.Telekommunikationsart</a>.<br/>
    /// <u><b>Telecommunication type codes:</b></u> version <b>2.2</b> from <a href="https://www.xrepository.de/api/xrepository/urn:xoev-de:xjustiz:codeliste:gds.telekommunikationsart_2.2:technischerBestandteilGenericode">XRepository: Code lists GDS.TelecommunicationType</a>.
    /// <br/><br/>
    /// <b>Änderung zur Vorversion:</b> Keine. Dies ist die erste Version.<br/>
    /// <b>Changelog:</b> None. This is the first version.
    /// </summary>
    public static IVersionedCodeList<TelekommunikationsartCode> V2_2 { get; } = new VersionedCodeList<TelekommunikationsartCode>("2.2")
    {
        TelekommunikationsartCode.Email,
        TelekommunikationsartCode.Fax,
        TelekommunikationsartCode.Internet,
        TelekommunikationsartCode.Mobiltelefon,
        TelekommunikationsartCode.Pager,
        TelekommunikationsartCode.Sonstige,
        TelekommunikationsartCode.Telefon,
        TelekommunikationsartCode.RechtsverbindlicherElektronischerKommunikationsweg,
    };

    public static string Uri => "urn:xoev-de:xjustiz:codeliste:gds.telekommunikationsart";

    public static IVersionedCodeList<TelekommunikationsartCode> LatestList => V2_2;
}

File: sdk/java/src/main/java/de/xjustiz/core/models/TelekommunikationsartCodeLists.java
package de.xjustiz.core.models;


public final class TelekommunikationsartCodeLists {

    private TelekommunikationsartCodeLists() {}

    public static final IVersionedCodeList<TelekommunikationsartCode> V2_2;

    static {
        V2_2 = new VersionedCodeList<>("2.2");
        initV2_2_0();
    }

    private static void initV2_2_0() {
        V2_2.add(TelekommunikationsartCode.Email);
        V2_2.add(TelekommunikationsartCode.Fax);
        V2_2.add(TelekommunikationsartCode.Internet);
        V2_2.add(TelekommunikationsartCode.Mobiltelefon);
        V2_2.add(TelekommunikationsartCode.Pager);
        V2_2.add(TelekommunikationsartCode.Sonstige);
        V2_2.add(TelekommunikationsartCode.Telefon);
        V2_2.add(TelekommunikationsartCode.RechtsverbindlicherElektronischerKommunikationsweg);
    }

}